File: Bai_24/main.py
# Hàm kiểm tra số nguyên tố
def is_prime(n):
    if n <= 1:
        return False
    for i in range(2, int(n**0.5) + 1):
        if n % i == 0:
            return False
    return True

# Đọc dữ liệu từ file input
with open('DOANNT.INP', 'r') as file:
    Q = int(file.readline().strip())
    results = []

    # Xử lý từng truy vấn
    for _ in range(Q):
        L, R = map(int, file.readline().split())
        count = 0
        for num in range(L + 1, R):
            if is_prime(num):
                count += 1
        results.append(count)

# Ghi kết quả vào file output
with open('DOANNT.OUT', 'w') as file:
    for result in results:
        file.write(str(result) + '\n')

File: Bai_24/main.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <cmath>

// Function to check if a number is prime
bool isPrime(int n)
{
    if (n <= 1)
    {
        return false;
    }
    for (int i = 2; i <= sqrt(n); ++i)
    {
        if (n % i == 0)
        {
            return false;
        }
    }
    return true;
}

int main()
{
    std::ifstream infile("DOANNT.INP");
    std::ofstream outfile("DOANNT.OUT");

    int Q;
    infile >> Q;
    std::vector<int> results;

    for (int q = 0; q < Q; ++q)
    {
        int L, R;
        infile >> L >> R;
        int count = 0;
        for (int num = L + 1; num < R; ++num)
        {
            if (isPrime(num))
            {
                count++;
            }
        }
        results.push_back(count);
    }

    for (int result : results)
    {
        outfile << result << "\n";
    }

    infile.close();
    outfile.close();

    return 0;
}
